compiler/next/include/chpl/types/RealType.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_TYPES_REALTYPE_H
+#define CHPL_TYPES_REALTYPE_H
+
+#include "chpl/types/PrimitiveType.h"
+
+namespace chpl {
+namespace types {
+
+
+/**
+  This class represents a real type, e.g. `real` or `real(32)`.
+ */
+class RealType : public PrimitiveType {
+ private:
+  RealType(int bitwidth)
+    : PrimitiveType(typetags::RealType, bitwidth)
+  { }
+
+  bool contentsMatchInner(const Type* other) const override {
+    return primitiveTypeContentsMatchInner((PrimitiveType*) other);
+  }
+
+  void markUniqueStringsInner(Context* context) const override {
+    primitiveTypeMarkUniqueStringsInner(context);
+  }
+
+  static const owned<RealType>& getRealType(Context* context, int bitwidth);
+
+ public:
+  ~RealType() = default;
+
+  static const RealType* get(Context* context, int bitwidth);
+
+  int bitwidth() const override {
+    return bitwidth_;
+  }
+
+  const char* c_str() const override {
+    switch (bitwidth_) {
+      case 32:
+        return "real(32)";
+      case 64:
+          return "real(64)";
+      default:
+        assert(false && "real bit width case not handled");
```

Comment:
Actually these assertions are already there. The constructor itself is private. These types are always created by a `get` function; e.g. `RealType::get`. These `get` functions include the assertion - see e.g. https://github.com/chapel-lang/chapel/pull/18121/files#diff-6a16b55dd7370459eaad5449212a3ae274da451ae615e83af2ff95280b4ef3f8R36

The `.c_str()` includes the duplicate assertion just in case the code becomes out of date between those two places.